src/amoc_metrics.erl

```diff
@@ -1,66 +1,96 @@
+    Datapoints = maps:get(report, Opts, HistProps),
+    create_metric_and_maybe_subscribe(ExName, histogram, Reporter, Interval, Datapoints).
 
 -spec new_spiral(name()) -> ok.
-new_spiral(Name) when is_atom(Name) ->
-    new_spiral(Name, []).
+new_spiral(Name) ->
+    new_spiral(Name, default_reporting_opts()).
 
--spec new_spiral(name(), [proplists:property()]) -> ok.
-new_spiral(Name, Opts) when is_atom(Name) ->
-    ExName = [amoc, spirals, Name],
-    {Reporter, Interval} = reporter_and_interval(Opts),
-    ReportProps = proplists:get_value(report, Opts, [one, count]),
-    ok = exometer:new(ExName, spiral),
-    ok = exometer_report:subscribe(Reporter, ExName, ReportProps, Interval).
+-spec new_spiral(name(), reporting_opts()) -> ok.
+new_spiral(Name, Opts) ->
+    ExName = make_name(spiral, Name),
+    #{reporter := Reporter, interval := Interval} = Opts,
+    Datapoints = maps:get(report, Opts, [one, count]),
+    create_metric_and_maybe_subscribe(ExName, spiral, Reporter, Interval, Datapoints).
 
 -spec new_counter(name()) -> ok.
 new_counter(Name) when is_atom(Name) ->
-    new_counter(Name, []).
+    new_counter(Name, default_reporting_opts()).
 
--spec new_counter(name(), [proplists:property()]) -> ok.
+-spec new_counter(name(), reporting_opts()) -> ok.
 new_counter(Name, Opts) when is_atom(Name) ->
     ExName = [amoc, counters, Name],
-    {Reporter, Interval} = reporter_and_interval(Opts),
-    ReportProps = proplists:get_value(report, Opts, [value]),
-    ok = exometer:new(ExName, counter),
-    ok = exometer_report:subscribe(Reporter, ExName, ReportProps, Interval).
+    #{reporter := Reporter, interval := Interval} = Opts,
+    Datapoints = maps:get(report, Opts, [value]),
+    create_metric_and_maybe_subscribe(ExName, counter, Reporter, Interval, Datapoints).
 
 -spec update_hist(name(), integer()) -> ok.
-update_hist(Name, Value) when is_atom(Name) ->
-    ExName = [amoc, histograms, Name],
+update_hist(Name, Value) ->
+    ExName = make_name(histogram, Name),
     exometer:update(ExName, Value).
 
 -spec update_spiral(name(), integer()) -> ok.
-update_spiral(Name, Value) when is_atom(Name) ->
-    ExName = [amoc, spirals, Name],
+update_spiral(Name, Value) ->
+    ExName = make_name(spiral, Name),
     exometer:update(ExName, Value).
 
 -spec update_counter(name(), integer()) -> ok.
-update_counter(Name, Value) when is_atom(Name) ->
-    ExName = [amoc, counters, Name],
+update_counter(Name, Value) ->
+    ExName = make_name(counter, Name),
     exometer:update(ExName, Value).
 
--spec reporter_and_interval([proplists:property()]) ->
-    {module(), non_neg_integer()}.
-reporter_and_interval(Opts) ->
-    Reporter = proplists:get_value(reporter, Opts, exometer_report_graphite),
-    Interval = proplists:get_value(interval, Opts, 5000),
-    {Reporter, Interval}.
+-spec messages_spiral_name() -> name().
+messages_spiral_name() ->
+    [counters, messages_sent].
```

Comment:
The `counters` prefix suggests it's a counter but it's a spiral. I think it's confusing.

One of the possible solutions would be prefixing all names with their type and enforcing the pattern `[amoc, Type, ...]` where `Type` is `histogram`, `spiral` or `counter` (I don't think the plural makes much sense here). The open question is whether `...` should be one atom or a list of atoms. We could also get rid of the type prefix completely.

I think the current situation is inconsistent as some prefixes are metric types (`histograms`, `spirals`, `counters`) and some are not (`times`).